src/compass/utils/h5_helpers.py

```diff
@@ -120,15 +127,28 @@ def init_geocoded_dataset(grid_group, dataset_name, geo_grid, dtype,
         output_kwargs['compression_opts'] = output_cfg.compression_level
         output_kwargs['shuffle'] = output_cfg.shuffle
 
+    # Shape of dataset is defined by the geo grid
     shape = (geo_grid.length, geo_grid.width)
+
+    # Determine fill value of dataset to either correctly init empty dataset
+    # and/or populate dataset attribute
+    _fill_val = determine_fill_value(dtype, fill_val)
+
+    # If data is None, create dataset to specified parameters and fill with
+    # specified fill value. If data is not None, create a dataset with
+    # provided data.
     if data is None:
-        cslc_ds = grid_group.require_dataset(dataset_name, dtype=dtype,
-                                             shape=shape, **output_kwargs)
+        # Create a dataset with shape and a fill value from above
+        cslc_ds = data_group.require_dataset(dataset_name, dtype=dtype,
+                                             shape=shape, fillvalue=_fill_val,
+                                             **output_kwargs)
     else:
-        cslc_ds = grid_group.create_dataset(dataset_name, data=data,
+        # Create a dataset with provided data
+        cslc_ds = data_group.create_dataset(dataset_name, data=data,
                                             **output_kwargs)
 
     cslc_ds.attrs['description'] = description
+    cslc_ds.attrs['fill_value'] = _fill_val
```

Comment:
Any reason to use `fill_value` instead of `_FillValue` like is [described here in the cf conventions](http://cfconventions.org/cf-conventions/cf-conventions.html#missing-data) ?

Reply:
> Any reason to use `fill_value` instead of `_FillValue` like is [described here in the cf conventions](http://cfconventions.org/cf-conventions/cf-conventions.html#missing-data) ?

I failed to take CF conventions into account. Thank you for pointing me in the right direction.